compiler/fuzz/fuzz_targets/fuzz_parser.rs

```diff
@@ -0,0 +1,93 @@
+#![no_main]
+
+use std::collections::HashMap;
+
+use candy::{
+    compiler::{hir, mir_to_lir::MirToLir, TracingConfig},
+    database::Database,
+    module::{InMemoryModuleProvider, Module, ModuleKind, Package},
+    vm::{
+        context::{DbUseProvider, RunForever},
+        tracer::dummy::DummyTracer,
+        Closure, ExecutionResult, Status, Struct, Vm,
+    },
+};
+use lazy_static::lazy_static;
+use libfuzzer_sys::fuzz_target;
+
+const TRACING: TracingConfig = TracingConfig::off();
+lazy_static! {
+    static ref PACKAGE: Package = Package::User("/".into());
+    static ref MODULE: Module = Module {
+        package: PACKAGE.clone(),
+        path: vec!["fuzzer".to_string()],
+        kind: ModuleKind::Code,
+    };
+}
+
+fuzz_target!(|data: &[u8]| {
+    let Ok(source_code) = String::from_utf8(data.to_vec()) else {
+        return;
+    };
+
+    let module_provider = InMemoryModuleProvider::for_modules::<String>(HashMap::new());
+    let mut db = Database::new(Box::new(module_provider));
+
+    db.did_open_module(&MODULE, source_code.as_bytes().to_owned());
```

Comment:
```suggestion
    let mut module_provider = InMemoryModuleProvider::default();
    module_provider.add(&MODULE, data.to_vec());
    let db = Database::new(Box::new(module_provider));

```

This version is a bit shorter, more explicit, and also tests our parser against invalid UTF-8